### ISSUE TITLE
Improve HSQLDB Dual table implementation

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#8888]
jOOQ currently uses a quite cumbersome `DUAL` emulation in HSQLDB:

```sql
select 1 as dual from information_schema.system_users limit 1
```

It might be better to use

```sql
values(1) dual(dual)
```

This is possible in HSQLDB 2.5. Perhaps, older versions do not support this syntax yet (either the `VALUES` constructor, or the derived column list)

**Comment by knutwannheden** [comment]
Yes, this would indeed be nice. Also supporting `VALUES` in other contexts (rather than using `UNION ALL`) would also be nice, but I guess we have an issue for that already.

**Comment by lukaseder** [comment]
What do you have in mind here? `org.jooq.impl.Values` already generates native `VALUES()` for HSQLDB, not `UNION ALL`...

**Comment by knutwannheden** [comment]
Right, I got confused by a test utility we have which generates `UNION ALL` queries. I will consider rewriting this to use `DSL#row()` and `DSL#values()` instead.

**Comment by lukaseder** [comment]
There are bugs and missing features with `DSL.values()`, unfortunately, e.g. 

- https://github.com/jOOQ/jOOQ/issues/5851
- https://github.com/jOOQ/jOOQ/issues/6003
- https://github.com/jOOQ/jOOQ/issues/7098
- https://github.com/jOOQ/jOOQ/issues/7897

**Comment by knutwannheden** [comment]
Just came across this again. The current emulation is actually:

```sql
(select 1 as dual from information_schema.system_users limit 1) as dual
```

For HSQLDB I don't even think we need the alias and could reduce it to the following:

```sql
(values(1))
```

Note: This also works with rather old versions like HSQLDB 2.2.9.

**Comment by lukaseder** [comment]
While the alias is not documented, there might be some cases where users already rely on it, e.g. when using `select(asterisk())`. Better not change the alias without making the table configurable: #6023

**Comment by lukaseder** [comment]
> Just came across this again

Curious: How did you "come across this again"? :) Some trouble you encountered with the existing emulation?

**Comment by knutwannheden** [comment]
> While the alias is not documented, there might be some cases where users already rely on it, e.g. when using `select(asterisk())`. Better not change the alias without making the table configurable: #6023

OK. Since the alias in `Dual.java` isn't the same for all dialects (for one it is `dummy` and for some it is absent) I was assuming we might as well drop it when not required.

> Curious: How did you "come across this again"? :) Some trouble you encountered with the existing emulation?

Yes. I noticed it here: #9187.

**Comment by lukaseder** [comment]
> OK. Since the alias in Dual.java isn't the same for all dialects (for one it is dummy and for some it is absent) I was assuming we might as well drop it when not required.

I'm a big fan of not touching status quo, unless it is truly fixed. Changing or removing the alias would replace one undocumented behaviour by another undocumented behaviour, without adding value to users (or ourselves), yet adding risk.

**Comment by lukaseder** [comment]
I'm going to implement this:

> ```sql
> values(1) dual(dual)
> ```

We don't have to re-use the `DSL.values()` API if there are bugs. Just generate `K_VALUES` and the other syntax explicitly, as we're doing for other dialects in `Dual.java`. All integration tests pass with the adaptation.
